src/server/requesthandler/base.py

```diff
@@ -0,0 +1,819 @@
+            else:
+                playready_la_url = urllib.unquote_plus(playready_la_url)
+            playready_version = self.request.params.get('playready_version')
+            if playready_version is not None:
+                playready_version = float(playready_version)
+        templates = TemplateFactory.get_singleton()
+        mspr = PlayReady(templates, la_url=playready_la_url, version=playready_version)
+        ck = ClearKey(templates)
+        marlin = Marlin(templates)
+        rv = {
+            'playready': {
+                'cenc': mspr.generate_pssh,
+                'laurl': playready_la_url,
+                'pro': mspr.generate_pro,
+                'moov': mspr.generate_pssh,
+                'scheme_id': mspr.dash_scheme_id(),
+                'version': playready_version,
+            },
+            'marlin': {
+                'MarlinContentIds': True,
+                'laurl': marlin_la_url,
+                'scheme_id': marlin.dash_scheme_id(),
+            },
+            'clearkey': {
+                'scheme_id': ck.dash_scheme_id(),
+                'laurl': self.generate_clearkey_license_url(),
+                'cenc': ck.generate_pssh,
+                'moov': ck.generate_pssh,
+            }
+        }
+        if playready_version == 1.0:
+            # PlayReady v1.0 (PIFF) mode only allows an mspr:pro element
+            rv['playready']['cenc'] = None
+            rv['playready']['moov'] = None
+        drms = self.request.params.get('drm')
+        if drms is None or drms == 'all':
+            return rv
+        d = {}
+        for name in drms.split(','):
+            try:
+                if '-' in name:
+                    parts = name.split('-')
+                    name = parts[0]
+                    d[name] = utils.objects.pick_items(
+                        rv[name],
+                        ['MarlinContentIds', 'laurl', 'scheme_id', 'version'])
+                    for p in parts[1:]:
+                        d[name][p] = rv[name][p]
+                else:
+                    d[name] = rv[name]
+            except KeyError:
+                pass
+        return d
+
+    def generateSegmentList(self, representation):
+        # TODO: support live profile
+        rv = ['<SegmentList timescale="%d" duration="%d">' %
+              (representation.timescale, representation.mediaDuration)]
+        first = True
+        for seg in representation.segments:
+            if first:
+                rv.append(
+                    '<Initialization range="{start:d}-{end:d}"/>'.format(start=seg.pos, end=seg.pos + seg.size - 1))
+                first = False
+            else:
+                rv.append('<SegmentURL mediaRange="{start:d}-{end:d}"/>'.format(
+                    start=seg.pos, end=seg.pos + seg.size - 1))
+        rv.append('</SegmentList>')
+        return '\n'.join(rv)
+
+    def generateSegmentDurations(self, representation):
```

Comment:
move into Representation class. Maybe change from a string to a context for a template?